lib/phoenix/endpoint.ex

```diff
@@ -858,8 +860,30 @@ defmodule Phoenix.Endpoint do
     * `:code_reloader` - enable or disable the code reloader. Defaults to your
       endpoint configuration
 
+    * `connect_info` - a list of keys that represent data to be copied from
+      the transport to be made available in the user socket `connect/3` callback.
+      
+      The valid keys are:
+
+        * `:peer_data` - the result of `Plug.Conn.get_peer_data/1`.
+        * `:x_headers` - all request headers that have an "x-" prefix.
+        * `:uri` - a `%URI{}` with information from the conn.
+
+      For example:
+
+        connect_info: [:peer_data, :x_headers, :uri]
+
+  ## Websocket configuration
+
+  The following configuration applies only to `:websocket`.
```

Comment:
Maybe end this (and the next one) using a semicolon?

I can do this in a different PR: some items in the lists end with dot but others do not :)

Reply:
I can definitely address all those formatting issues in a different PR. There are also “inconsistencies” in Transport :)